06.wc/wc.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options = ARGV.getopts('l', 'w', 'c')
+  options = { 'l' => true, 'w' => true, 'c' => true } if options.values.none?
+  options = options.transform_keys(&:to_sym)
+  files = make_source_files
+  outputs = count_outputs(files)
+  if outputs.size > 1
+    total = count_total(outputs)
+    outputs << total
+  end
+  print_outputs(options, outputs)
+end
+
+def make_source_files
+  if ARGV.size >= 1
+    ARGV.map do |file|
+      {
+        file_contents: File.read(file),
+        file_name: file
+      }
+    end
+  else
+    [] << { file_contents: $stdin.read }
```

Comment:
Rubyの`<<`は要素が追加された配列自身を返しますが、実際の使われ方としては戻り値を利用せず「要素を追加するだけ」で終わることが多いです。
なので、`<<`の戻り値をメソッドの`make_source_files `の戻り値に使うのは若干違和感があります。

そもそもここは

```suggestion
    [{ file_contents: $stdin.read }]
```

もしくは

```suggestion
    [file_contents: $stdin.read]
```

だけで良かったりします。